### ISSUE TITLE
Browser Notification type definition incorrect

### ISSUE THREAD

**Issue by chris-ls** [Microsoft/TypeScript#14701]
**TypeScript Version:**  2.2.1

Looking at lib.es6.d.ts we have
```
interface Notification extends EventTarget {
    // ...
    readonly permission: string;
    // ...
}
```
and
```
declare var Notification: {
    prototype: Notification;
    new(title: string, options?: NotificationOptions): Notification;
    requestPermission(callback?: NotificationPermissionCallback): Promise<string>;
}
```

The `permission` string is a static property on the `Notification` class.
i.e. We should have:
```
declare var Notification: {
    prototype: Notification;
    readonly permission: string;
    new(title: string, options?: NotificationOptions): Notification;
    requestPermission(callback?: NotificationPermissionCallback): Promise<string>;
}
```
and it should be removed from the `Notification` Interface definition.

**Comment by NNemec** [comment]
Just stumbled over the same issue a today...

**Comment by mhegazy** [comment]
Seems we are also missing `maxActions` as a static property on `Notification`

**Comment by mhegazy** [comment]
PRs welcomed. You can find more information about contributing lib.d.ts fixes at https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes.

**Comment by csgulyas** [comment]
I think this would require modifying `browser.webidl.xml` in the [TSJS-lib-generator](https://github.com/Microsoft/TSJS-lib-generator/tree/master/inputfiles) which the readme says don't, because it comes from Edge

**Comment by NNemec** [comment]
Does that mean that the actual bug is in Edge? Any hope to get it fixed there?
Or is the bug in Typescript using that file in a way that is not intended by Edge?

**Comment by csgulyas** [comment]
I'm not sure how `browser.webidl.xml` comes into existence, if it is a result of an extraction process from Edge, or Edge simply interprets it differently, but locally I tried to modify the file and re-generate `dom.generated.d.ts`.

I removed `permission` from the properties node and added it as a constant:
```xml
<interface name="Notification" extends="EventTarget">
	  <constants>
		<constant name="permission" type="NotificationPermission" />
	  </constants>
...
```

The output was more along the lines of what OP suggested
```TypeScript
interface Notification extends EventTarget {
    ...
    readonly permission: NotificationPermission;
}

declare var Notification: {
    prototype: Notification;
    new(title: string, options?: NotificationOptions): Notification;
    readonly permission: NotificationPermission;
    requestPermission(callback?: NotificationPermissionCallback): Promise<NotificationPermission>;
}
```
I don't think the build tool supports removing things from the interface while keeping it in the var declaration.

**Comment by kvetis** [comment]
@csgulyas Yes, you've identified the issue correctly. The `browser.webidl.xml` includes incorrect types. The files [comes from the Edge team](https://github.com/Microsoft/TSJS-lib-generator/issues/101#issuecomment-226980353). There is an effort to change this to WHATWG [currently underway](https://github.com/Microsoft/TSJS-lib-generator/pull/300). If I understand correctly, this would allow the typings to be generated from WHATWG instead of Edge standards.

That, I think, should fix this issue, altough there still is a lot of work ahead. **A huge thanks to people working on that.**

Also related to https://github.com/Microsoft/TypeScript/issues/3027

**Comment by Deathangel908** [comment]
Which version fixes this? `"typescript": "^2.9.1",`

> TS2339: Property 'permission' does not exist on type '{ new (title: string,
>  options?: NotificationOptions): Notification; prototype: Notification; requ...'.

**Comment by mhegazy** [comment]
Should be in `typescript@3.0`
